spec/memfs/dir_spec.rb

```diff
@@ -264,6 +264,18 @@ module MemFs
         expect(File.directory?('/new-folder')).to be true
       end
 
+      it 'sets directory permissions to default 0777' do
+        described_class.mkdir '/new-folder'
+        expect(File.stat('/new-folder').mode).to eq(0100777)
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.